### PR TITLE
docs: add documentation for hasAnyCheckRunCompleted built-in function

### LINKED ISSUE #228: has any check run completed built-in documentation
## Description
Add documentation for `$hasAnyCheckRunCompleted` built-in function.

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Adds documentation for the new `$hasAnyCheckRunCompleted` built-in function.

## Related issue

Closes #228 

## Checklist

- [x] I have performed a self-review of my code
- [x] I have run `yarn build` without any errors

## Code review and merge strategy (ship/show/ask)

**Ask**: this pull request requires a code review before merge

### CODE REVIEW

File: docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+**Description**:
+
+Checks if any check runs for a pull request have been completed with the provided conclusions.
+
+If the `check_runs_to_ignore` parameter is provided the checks with a name included in the parameter will be ignored.
+
+If the `check_conclusions` parameter is provided, any check run must be complete with one of the provided conclusions.
+
+The `check_conclusions` parameter can contain one of `action_required`, `cancelled`, `failure`, `neutral`, `success`, `skipped`, `stale` or `timed_out`.
```
Comment (shay2025) [inline docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx:17]: WDYT?
```suggestion
Checks whether any check runs for a pull request have been completed with the provided conclusions.

If the `check_runs_to_ignore` parameter is provided, any check runs that have a name included in the parameter will be ignored.

If the `check_conclusions` parameter is provided, only consider check runs that have been completed with one of the provided conclusions will be considered.

The `check_conclusions` parameter accepts the following values: `action_required`, `cancelled`, `failure`, `neutral`, `success`, `skipped`, `stale`, and `timed_out`.
```

File: docs/guides/built-ins/functions/hasAnyCheckRunCompleted/example.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+api-version: reviewpad.com/v3.x
+
+workflows:
+  - name: check-run-failure
+    if:
+      - $hasAnyCheckRunCompleted([], ["failure"])
+    then:
+      - $comment("All checks must complete successfully")
```
Comment (shay2025) [inline docs/guides/built-ins/functions/hasAnyCheckRunCompleted/example.yml:8]: WDYT?
```suggestion
      - $review("REQUEST_CHANGES", "Pull request checks are failing. Please fix them.")
```

**Review by shay2025 — CHANGES_REQUESTED** [review]
Some suggestions

**Review by shay2025 — APPROVED** [review]

File: docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+# This slug is a workaround for the links on this page to work.
+slug: "/guides/built-ins-haveallchecksruncompleted"
+---
```
Comment (ferreiratiago) [inline docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx:4]: Please remove the work around. This is only needed for pages with links.

File: docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+:::caution
+
+This built-in is not available for issues.
+
+:::
+
+**Parameters**:
+
+| Parameter                                | Type       | Description                                             |
```
Comment (ferreiratiago) [inline docs/guides/built-ins/functions/hasAnyCheckRunCompleted/index.mdx:27]: Please check the formatting here.

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**